src/app/pages/question/question.component.ts

```diff
@@ -1,36 +1,63 @@
 import { Component, Input, OnInit } from '@angular/core';
 import { UIKitModule } from '../../ui-kit/ui-kit.module';
 import { ActivatedRoute } from '@angular/router';
-import { QUIZ_DATA } from '../../test-data/quiz';
 import { CommonModule } from '@angular/common';
+import { QuestionModel } from '../../services/model/question.model';
+import { StoreService } from '../../services/store.service';
+import { BehaviorSubject, combineLatest, map, Observable, take, tap } from 'rxjs';
 
 @Component({
   standalone: true,
   selector: 'app-quiz',
   templateUrl: './question.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizComponent implements OnInit {
+export class QuestionComponent implements OnInit {
   @Input() quiz: string = '';
-  quiz_data = QUIZ_DATA;
-  currentIndex: number = 0;
-  
-  constructor(private route: ActivatedRoute) {}
+
+  questions$!: Observable<QuestionModel[]>;
+  isLoading$!: Observable<boolean>;
+  currentIndex$ = new BehaviorSubject<number>(0);
+  currentQuestion$!: Observable<QuestionModel>;
+
+  constructor(private route: ActivatedRoute, private storeService: StoreService) {}
 
   nextQuestion(): void {
-    if (this.currentIndex < this.quiz_data.length - 1) {
-      this.currentIndex++;
+    const currentIndex = this.currentIndex$.value;
+    if (currentIndex < 9) {
```

Comment:
What with the 9 here? What does it mean?

Reply:
I mean what 9 stands for? If we have 10 questions as maximum we need to specify that somewhere as a const (in model, util or type file) and export it